metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvesterImpl.java

```diff
@@ -34,6 +42,55 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(HttpHarvesterImpl.class);
 
+  @Override
+  public void harvestRecords(InputStream inputStream, CompressedFileExtension compressedFileType,
+          Consumer<ArchiveEntry> action) throws HarvesterException {
```

Comment:
The existing harvesting functionality works on a temporary file, not on an input stream. It is due to the fact that the zip file might be large. So the idea is to first save it to the file system and then we can do all operations on it without risking memory issues. Of course for our test application this is not really a concern, but it is easiest to just do this so that we can use the functionality already implemented below. And you never know...